myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java

```diff
@@ -92,12 +97,39 @@ public synchronized void storeMyriadState(StoreContext sc) throws Exception {
 
     LOGGER.debug("Storing state information for Myriad at: " + myriadStatePath);
     try {
-      updateFile(myriadStatePath, sc.toSerializedContext().toByteArray());
+      reflectedUpdateFile(myriadStatePath, sc.toSerializedContext().toByteArray());
     } catch (Exception e) {
       LOGGER.error("State information for Myriad could not be stored at: " + myriadStatePath, e);
     }
   }
 
+  private synchronized void getCachedUpdateFileMethod() {
+    if (cachedUpdateFileMethod == null) {
+      Method[] methods = this.getClass().getDeclaredMethods();
+      for (Method m : methods) {
+        if (m.getName().equals("updateFile")) {
+          cachedUpdateFileMethod = m;
+          break;
+        }
+      }
+    }
+  }
+
+  protected void reflectedUpdateFile(Path outputPath, byte[] data) throws InvocationTargetException, IllegalAccessException {
+    if (cachedUpdateFileMethod == null) {
```

Comment:
After looking over this again, I think I can just initialize once in `InitInternal(...)` and drop the lazy eval business.  Solves the null checks much cleaner.